src/main/scala/beam/agentsim/agents/freight/input/GenericFreightReader.scala

```diff
@@ -178,6 +179,13 @@ class GenericFreightReader(
     allPlans: Map[Id[PayloadPlan], PayloadPlan],
     vehicleTypes: Map[Id[BeamVehicleType], BeamVehicleType]
   ): IndexedSeq[FreightCarrier] = {
+    val freightVehicleTypes: Map[Id[BeamVehicleType], BeamVehicleType] = {
+      config.vehicleTypesFilePath match {
+        case Some(filePath) => vehicleTypes ++ readBeamVehicleTypeFile(filePath)
+        case None           => vehicleTypes
+      }
+    }
```

Comment:
To improve readability we could remove the outer curly braces. Though it's up to you.